matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/aggregation/livelocation/LiveLocationAggregationProcessor.kt

```diff
@@ -137,5 +142,16 @@ internal class LiveLocationAggregationProcessor @Inject constructor(
         }
     }
 
+    private fun deactivateAllPreviousBeacons(realm: Realm, roomId: String, userId: String, currentEventId: String) {
+        LiveLocationShareAggregatedSummaryEntity
+                .findActiveLiveInRoomForUser(
+                        realm = realm,
+                        roomId = roomId,
+                        userId = userId
+                )
+                .filterNot { it.eventId == currentEventId }
```

Comment:
do we expect to have many events per user/room? it might be safer to do the filtering within the database rather than on the query results (I'm imagining a user with years worth of history creating big in memory lists)

Reply:
I see you point. Here we only query the active lives, meaning we should not have a lot of items in the list. But I agree, it is safer to integrate the filter inside the query.
In the future, I guess we may allow multiple lives per user (one per device).

Reply:
See [066c540](https://github.com/vector-im/element-android/pull/6129/commits/066c540eb70bef15b3fb59fcc72ea283117e0934)